### ISSUE TITLE
[REQUEST] ELO system

### ISSUE THREAD

**Issue by Haxzer** [PhlexPlexico/G5-Bot#10]
If it'd be possible to add an ELO system or score system (like Elo Bot) https://github.com/tournament-kings/legacy_elo that automatically gets and saves a local Elo for use in gather groups :)

**Comment by PhlexPlexico** [comment]
That's an interesting idea... However I could only see this going two ways; one would be a direct integration with (end-of-life) Get5-web, where it would simply get the values of whoever is queuing via a leaderboard query, much like it's already done since it's a relatively quick query. But then that adds more dependencies for features to be added.

The next option would be to provide an interface from CS -> Discord. I know plugins do exist on servers for this so it would then be a requirement/need a separate plugin, which is personally more work than I would want to put into this bot, considering half of it is now no longer in development/outdated AF.

I would much prefer something to be a bit more automated than what you've linked as well, since that can make up some human error/takes time to enter stats.

At the end of the day, this bot was just meant for organizing ten-mans within Discord just so users wouldn't have to navigate through the site in an attempt to set everything up themselves. If you would want to see the stats/ratings of people, the solution would be to navigate to the webpanel to see it. Otherwise, I wanted to keep this bot as simple as possible and just let people choose teams themselves.

**Comment by Haxzer** [comment]
I was more thinking just a database for the bot with a default "Score" or ELO of say 700, then if team1 wins, on discord, team1 gets a score +30 or something, and team2 gets a -30. As I believe the bot is "linked" with the discord ID's of the players

EDIT: What I mean is not to directly integrate it with the web-panel, just get a win/loss based rating score probably

**Comment by PhlexPlexico** [comment]
Hmm true. I think that would make more sense in... Actually I never spun off a repo for a more generalized bot. However, this could be something I look into, even as an optional requirement and include it in the setup instructions of the bot. However, since CS:GO lacks rich presence, it's difficult to provide a truly automated win/loss system, which is something I would like to do instead of having a user/mod input the win/loss because some people would definitely abuse that, especially if they get thrashed, hah.

**Comment by Haxzer** [comment]
Agreed that the truly automated win/loss system would be the best, maybe try to integrate the database of the get5-web so that you can automatically get the score (mostly for win/loss) to like, just read the score
